src/pipeline.ts

```diff
@@ -489,14 +541,507 @@ export async function scanRssFeeds() {
+						message: `${progress} Injected ${filePathLog} using stalled source, you will need to resume or remove from client - ${chalk.green(injectionResult)}`,
+					});
+					inClient = true;
+					injected = true;
+				} else {
+					logger.warn({
+						label: Label.INJECT,
+						message: `${progress} Failed to inject ${filePathLog} using stalled source - ${chalk.yellow(injectionResult)}`,
+					});
+				}
+			} else {
+				logger.warn({
+					label: Label.INJECT,
+					message: `${progress} Failed to link files for ${filePathLog}, ${linkedFilesRootResult.unwrapErr()} - ${chalk.yellow(injectionResult)}`,
+				});
+			}
+		}
+	}
+	if (inClient && !injected) {
+		if (linkedNewFiles) {
+			logger.info({
+				label: Label.INJECT,
+				message: `${progress} Rechecking ${filePathLog} as new files were linked - ${chalk.green(injectionResult)}`,
+			});
+			toRecheck.add(meta.infoHash);
+		} else {
+			logger.warn({
+				label: Label.INJECT,
+				message: `${progress} No new files linked for ${filePathLog}, resume or remove from client - ${chalk.yellow(injectionResult)}`,
+			});
+		}
+	}
+	return injected;
+}
+
+export async function injectSavedTorrents() {
+	const { flatLinking, injectDir, linkDir, outputDir } = getRuntimeConfig();
+	const targetDir = injectDir ?? outputDir;
+	const targetDirLog = chalk.bold.magenta(targetDir);
+
+	if (injectDir) {
+		// injectDir defined only by `cross-seed inject`, regardless if they specify it
+		logger.warn({
+			label: Label.INJECT,
+			message: `Manually injecting torrents performs minimal filtering which slightly increases chances of false positives, see the docs for more info`,
+		});
+	}
+	try {
+		if (!fs.statSync(targetDir).isDirectory()) {
+			logger.error({
+				label: Label.INJECT,
+				message: `${targetDirLog} is not a directory`,
+			});
+			return;
+		}
+	} catch (e) {
+		logger.error({
+			label: Label.INJECT,
+			message: `Error accessing ${targetDirLog} - ensure it exists and is accessible (verify docker volumes if applicable)`,
+		});
+		logger.debug(e);
+		return;
+	}
+	const dirContents = await findAllTorrentFilesInDir(targetDir);
+	if (dirContents.length === 0) {
+		logger.info({
+			label: Label.INJECT,
+			message: `No torrent files found to inject in ${targetDirLog}`,
+		});
+		return;
+	}
+	logger.info({
+		label: Label.INJECT,
+		message: `Found ${chalk.bold.white(dirContents.length)} torrent file(s) to inject in ${targetDirLog}`,
+	});
+
+	const { searchees } = await findSearchableTorrents(Label.INJECT, {
+		useFilters: false,
+	});
+
+	const toDelete = new Set<string>();
+	const toRecheck = new Set<string>();
+
+	// Usually source got deleted or partial injection never completes
+	function shouldCleanUpTorrent(torrentFilePath: string) {
+		return fs.statSync(torrentFilePath).mtimeMs < Date.now() - ms("1 week");
+	}
+	function getTorrentFilePathLog(torrentFilePath: string) {
+		return chalk.bold.magenta(
+			torrentFilePath.replace(
+				/(?:\[)([a-z0-9]{40})(?:].torrent$)/i,
```

Comment:
are these non-capturing groups doing anything?
```suggestion
				/\[([a-z0-9]{40})].torrent$/i,
```